src/test/java/org/opensearch/flowframework/workflow/CreateIngestPipeline/CreateIngestPipelineStepTests.java

```diff
@@ -0,0 +1,155 @@
+            public Map<String, Object> getContent() {
+                return Map.of(
+                    "id",
+                    "pipelineId",
+                    "description",
+                    "some description",
+                    "type",
+                    "text_embedding",
+                    "model_id",
+                    "model_id",
+                    "input_field_name",
+                    "inputField",
+                    "output_field_name",
+                    "outputField"
+                );
+            }
+
+            @Override
+            public Map<String, String> getParams() {
+                return Map.of();
+            }
+        };
+
+        // Set output data to returned pipelineId
+        outpuData = new WorkflowData() {
+
+            @Override
+            public Map<String, Object> getContent() {
+                return Map.of("pipelineId", "pipelineId");
+            }
+
+            @Override
+            public Map<String, String> getParams() {
+                return Map.of();
+            }
+        };
+
+        client = mock(Client.class);
+        adminClient = mock(AdminClient.class);
+        clusterAdminClient = mock(ClusterAdminClient.class);
+
+        when(client.admin()).thenReturn(adminClient);
+        when(adminClient.cluster()).thenReturn(clusterAdminClient);
+    }
+
+    public void testCreateIngestPipelineStep() throws InterruptedException, ExecutionException {
+
+        CreateIngestPipelineStep createIngestPipelineStep = new CreateIngestPipelineStep(client);
+
+        ArgumentCaptor<ActionListener> actionListenerCaptor = ArgumentCaptor.forClass(ActionListener.class);
+        CompletableFuture<WorkflowData> future = createIngestPipelineStep.execute(List.of(inputData));
+
+        assertFalse(future.isDone());
+
+        // Mock put pipeline request execution and return true
+        verify(clusterAdminClient, times(1)).putPipeline(any(PutPipelineRequest.class), actionListenerCaptor.capture());
+        actionListenerCaptor.getValue().onResponse(new AcknowledgedResponse(true));
+
+        assertTrue(future.isDone());
+        assertEquals(outpuData.getContent(), future.get().getContent());
+    }
+
+    public void testCreateIngestPipelineStepFailure() {
+
+        CreateIngestPipelineStep createIngestPipelineStep = new CreateIngestPipelineStep(client);
+
+        ArgumentCaptor<ActionListener> actionListenerCaptor = ArgumentCaptor.forClass(ActionListener.class);
+        CompletableFuture<WorkflowData> future = createIngestPipelineStep.execute(List.of(inputData));
+
+        assertFalse(future.isDone());
+
+        // Mock put pipeline request execution and return false
+        verify(clusterAdminClient, times(1)).putPipeline(any(PutPipelineRequest.class), actionListenerCaptor.capture());
+        actionListenerCaptor.getValue().onFailure(new Exception());
+
+        assertTrue(future.isDone());
+        assertThrows(Exception.class, () -> future.get());
```

Comment:
This is not the same `Exception` you threw on line 122.  This is an `ExecutionException` with the line 122 exception as the cause.

Recommend testing `ExecutionException` here at a minimum.  Optionally add a message to the line 122 exception and test the getCause message matches.